internal/validation/constraints.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package validation
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+// Constraint is a constraint on a document or a field of a document.
+type Constraint struct {
+	// Satisfied returns true if the constraint is satisfied.
+	Satisfied func() (valid bool, err error)
```

Comment:
Would there be any downsides to having `Satisifed` return just an error, instead of bool + error?